### PR TITLE
Updated fees

### PR DESCRIPTION
**Pull request by tomasmik**
Requires: https://github.com/mysteriumnetwork/transactor/pull/416

### CODE REVIEW

**Review by n10ty — APPROVED** [review]

File: identity/registry/transactor.go [hunk 1]
```diff
@@ -229,6 +229,33 @@ func (t *Transactor) FetchStakeDecreaseFee(chainID int64) (FeesResponse, error)
...
+	DecreaseStake *big.Int  `json:"decreaseStake"`
+	Settle        *big.Int  `json:"settle"`
+	Register      *big.Int  `json:"register"`
+	ValidUntil    time.Time `json:"valid_until"`
+}
+
+// FetchCombinedFees fetches current transactor fees.
+func (t *Transactor) FetchCombinedFees(chainID int64) (CombinedFeesResponse, error) {
+	f := CombinedFeesResponse{}
+	req, err := requests.NewGetRequest(t.endpointAddress, fmt.Sprintf("fee/%v", chainID), nil)
```
Comment (Guillembonet) [inline identity/registry/transactor.go:250]: maybe use the cache, just to make sure this endpoint is not overused in node ui and results in a lot of calls to transactor?

File: tequilapi/endpoints/transactor.go [hunk 2]
```diff
@@ -119,6 +120,54 @@ func NewTransactorEndpoint(
...
+		c.Error(err)
+	}
+
+	hermes, err := te.addressProvider.GetActiveHermes(chainID)
+	if err != nil {
+		c.Error(apierror.Internal("Failed to get active hermes", contract.ErrCodeActiveHermes))
+		return
+	}
+
+	hermesFeePerMyriad, err := te.promiseSettler.GetHermesFee(chainID, hermes)
```
Comment (Guillembonet) [inline tequilapi/endpoints/transactor.go:153]: fancy word :monocle_face:

**Review by Guillembonet — APPROVED** [review]

File: identity/registry/transactor.go (hunk 1 above)
Comment (tomasmik) [reply identity/registry/transactor.go:250]: I dont like the cache with time sensitive expiration for the fees. The cache also doesnt fit this struct so I would have to re-write how it works. a bit annoying.

Comment (Guillembonet) [reply identity/registry/transactor.go:250]: then we can just remove it for all I guess

File: tequilapi/endpoints/transactor.go [hunk 3]
```diff
@@ -119,6 +120,54 @@ func NewTransactorEndpoint(
...
+		Last:       contract.NewTransactorFees(&fees.Last),
+		ServerTime: fees.ServerTime,
+
+		HermesPercent: hermesPercent.StringFixed(4),
+	}
+
+	utils.WriteAsJSON(f, c.Writer)
+}
+
 // swagger:operation GET /transactor/fees FeesDTO
```
Comment (mdomasevicius) [inline tequilapi/endpoints/transactor.go:171]: should mark as deprecated perhaps

File: tequilapi/contract/transactor.go [hunk 4]
```diff
@@ -48,6 +49,36 @@ type FeesDTO struct {
 	DecreaseStakeTokens Tokens   `json:"decrease_stake_tokens"`
 }
 
+// CombinedFeesDTO all transactor fees.
+// swagger:model CombinedFeesDTO
+type CombinedFeesDTO struct {
```
Comment (mdomasevicius) [inline tequilapi/contract/transactor.go:54]: `CombinedFeesDTO` -> `CombinedFeesResponse` 

everything is DTO :)

File: tequilapi/endpoints/transactor.go [hunk 5]
```diff
@@ -850,6 +903,10 @@ func AddRoutesForTransactor(
 			transGroup.GET("/token/:token/reward", a.TokenRewardAmount)
 			transGroup.GET("/chain-summary", te.ChainSummary)
 		}
+		transGroupV2 := e.Group("/v2/transactor")
+		{
+			transGroupV2.GET("/fees", te.TransactorFees)
```
Comment (mdomasevicius) [inline tequilapi/endpoints/transactor.go:908]: looks like a bug

**Review by mdomasevicius — APPROVED** [review]
awkward API :)

**Pull request merged**